src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,229 @@
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant).hasSize(1);
+        assertThat(valueInApplicant.get(0).get("value").equals(Applicant.builder()
+            .party(ApplicantParty.builder()
+                .partyId(null)
+                .partyType(null)
+                .name(APPLICANT)
+                .address(Address.builder()
+                    .build())
+                .emailAddress(EmailAddress.builder()
+                    .email(EMAIL)
+                    .build())
+                .telephoneNumber(TelephoneNumber.builder()
+                    .build())
+                .mobileNumber(MobileNumber.builder()
+                    .build())
+                .jobTitle(null)
+                .pbaNumber(null)
+                .build())
+                .leadApplicantIndicator(null)
+            .build()));
+    }
+
+    @Test
+    void whenPartiallyFilledInOldApplicantAddressIsMigratedShouldReturnNewListStructureWithNullFields() {
+        Map<String, Object> data = new HashMap<>();
+        OldApplicant applicant = OldApplicant.builder()
+            .address(Address.builder()
+                .postcode(POSTCODE)
+                .county(COUNTY)
+                .build())
+            .build();
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant).hasSize(1);
+        assertThat(valueInApplicant.get(0).get("value").equals(Applicant.builder()
+            .party(ApplicantParty.builder()
+                .partyId(null)
+                .partyType(null)
+                .name(null)
+                .address(Address.builder()
+                    .addressLine1(null)
+                    .addressLine2(null)
+                    .addressLine3(null)
+                    .postcode(POSTCODE)
+                    .country(COUNTRY)
+                    .county(null)
+                    .build())
+                .emailAddress(EmailAddress.builder()
+                    .email(null)
+                    .build())
+                .telephoneNumber(TelephoneNumber.builder()
+                    .telephoneNumber(null)
+                    .build())
+                .mobileNumber(MobileNumber.builder()
+                    .telephoneNumber(null)
+                    .build())
+                .jobTitle(null)
+                .pbaNumber(null)
+                .build())
+            .leadApplicantIndicator(null)
+            .build()));
+    }
+
+    private Applicant newApplicantBuilder() {
```

Comment:
Wired name, sort of suggests that object builder is returned but no, surprise, complete instance is returned instead.

Reply:
👍